### PR TITLE
Try to not use junit('coverage.xml')

### PR DESCRIPTION
**Pull request by jirihnidek**
I created this PR to test some quick workaround.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -19,11 +18,25 @@ pipeline {
...
             sh readFile(file: 'jenkins/python3-tests.sh')
-            junit('coverage.xml')
-            // TODO: find the correct adapter or generate coverage tests that can be
-            //       parsed by an existing adapter:
-            //       https://plugins.jenkins.io/code-coverage-api/
-            // publishCoverage adapters: [jacocoAdapter('coverage.xml')]
+          }
+          post {
+            always {
+              archiveArtifacts allowEmptyArchive: true, artifacts: 'coverage.xml', fingerprint: true
```
Comment (Lorquas) [inline Jenkinsfile:24]: Saves the xml file itself so it can be downloaded and manipulated directly from the Artifacts tab

File: Jenkinsfile [hunk 2]
```diff
@@ -19,11 +18,25 @@ pipeline {
...
+                            autoUpdateHealth: false,
+                            autoUpdateStability: false,
+                            coberturaReportFile: 'coverage.xml',
+                            failNoReports: false,
+                            failUnhealthy: false,
+                            failUnstable: false,
+                            maxNumberOfBuilds: 10,
+                            onlyStable: false,
+                            sourceEncoding: 'ASCII',
+                            zoomCoverageChart: false])
```
Comment (Lorquas) [inline Jenkinsfile:36]: I picked a permissive set of options so it won't fail ever (we should decide what we want), but you get nice published reports now.  From the blue ocean click on "Go to Classic" (square with the right arrow) > Coverage report  .  The PR page in the classic view will also have  an overall graph.   This publisher also would make it worthwhile to start running the tests against the main branch once these PRs are merged so we could see longer term trends between PRs.

File: Jenkinsfile [hunk 3]
```diff
@@ -19,11 +18,25 @@ pipeline {
...
+              // https://www.jenkins.io/doc/pipeline/steps/junit/
+              junit allowEmptyResults: true, keepLongStdio: true, skipMarkingBuildUnstable: true, skipPublishingChecks: true, testResults: 'coverage.xml'
```
Comment (Lorquas) [inline Jenkinsfile:38]: I'm still trying to figure out why the junit decided to stop working, but I added some flags that make this non blocking.  If it turns out to be a plugin/jenkins problem we won't have to edit this line to get the junit tests section working again.

**Comment by cnsnyder** [comment]
Looks good to me!

**Pull request merged**